src/server/game/Garrison/Garrison.cpp

```diff
@@ -589,8 +580,8 @@ std::pair<std::vector<GarrMissionEntry const*>, std::vector<double>> Garrison::G
         if (missionEntry->RequiredFollowersCount > activeFolowerCount)
             continue;
 
-        if (missionEntry->RequiredItemLevel > averageFollowerILevel)
-            continue;
+        //if (missionEntry->RequiredItemLevel > averageFollowerILevel)
+        //    continue;
```

Comment:
Why commenting ?

Reply:
this's wrong, example: mission required 750 follower Item level for maching bias, but ur followers has just 730 ilvl - u still'll be able to send forllowers on this mission but with lower succes chanse

Reply:
```c#

		public static float ComputeFollowerBias(WrapperGarrisonFollower follower, int followerLevel, int followerItemLevel, int garrMissionID)
		{
			bool flag = (follower.Flags & 8) != 0;
			if (flag)
			{
				return 0f;
			}
			GarrMissionRec record = StaticDB.garrMissionDB.GetRecord(garrMissionID);
			if (record == null)
			{
				return 0f;
			}
			float num = MissionDetailView.ComputeFollowerLevelBias(follower, followerLevel, record.TargetLevel);
			num += MissionDetailView.ComputeFollowerItemLevelBias(follower, followerItemLevel, (int)record.GarrFollowerTypeID, (int)record.TargetItemLevel);
			return Mathf.Clamp(num, -1f, 1f);
		}

		public static int GarrisonFollower_GetMaxFollowerLevel(int garrFollowerTypeID)
		{
			return (int)StaticDB.garrFollowerLevelXPDB.GetRecordsWhere((GarrFollowerLevelXPRec rec) => (ulong)rec.GarrFollowerTypeID == (ulong)((long)garrFollowerTypeID)).Max((GarrFollowerLevelXPRec rec) => rec.FollowerLevel);
		}

		private static float ComputeFollowerItemLevelBias(WrapperGarrisonFollower follower, int followerItemLevel, int garrFollowerTypeID, int targetItemLevel)
		{
			float num = 0f;
			if (follower.FollowerLevel == MissionDetailView.GarrisonFollower_GetMaxFollowerLevel(garrFollowerTypeID))
			{
				int num2;
				int num3;
				MissionDetailView.GarrisonFollower_GetFollowerBiasConstants(follower, out num2, out num3);
				int num4 = (targetItemLevel <= 0) ? 600 : targetItemLevel;
				float num5 = (float)(followerItemLevel - num4);
				num5 /= (float)num3;
				num += num5;
			}
			return num;
		}

		private static float ComputeFollowerLevelBias(WrapperGarrisonFollower follower, int followerLevel, int targetLevel)
		{
			int num;
			int num2;
			MissionDetailView.GarrisonFollower_GetFollowerBiasConstants(follower, out num, out num2);
			float num3 = (float)(followerLevel - targetLevel);
			return num3 / (float)num;
		}

		private static float ComputeBiasValue(float par, float max, float bias)
		{
			if (bias < 0f)
			{
				return par + bias * par;
			}
			return par + bias * (max - par);
		}

		private static void GarrisonFollower_GetFollowerBiasConstants(WrapperGarrisonFollower follower, out int out_levelRangeBias, out int out_itemLevelRangeBias)
		{
			out_levelRangeBias = 1;
			out_itemLevelRangeBias = 1;
			GarrFollowerRec record = StaticDB.garrFollowerDB.GetRecord(follower.GarrFollowerID);
			if (record == null)
			{
				return;
			}
			GarrFollowerTypeRec record2 = StaticDB.garrFollowerTypeDB.GetRecord((int)record.GarrFollowerTypeID);
			if (record2 == null)
			{
				return;
			}
			out_levelRangeBias = (int)((record2.LevelRangeBias >= 1u) ? record2.LevelRangeBias : 1u);
			out_itemLevelRangeBias = (int)((record2.ItemLevelRangeBias >= 1u) ? record2.ItemLevelRangeBias : 1u);
		}
```

Reply:
example how it calcs

Reply:
Ok thanks for the infos